nixos/modules/services/web-apps/invidious.nix

```
@@ -7,17 +7,31 @@ let
 
   settingsFile = settingsFormat.generate "invidious-settings" cfg.settings;
 
+  finalHmacKeyFile =
+    if cfg.hmacKeyFile != null
+    then cfg.hmacKeyFile
+    else "/var/lib/invidious/hmac_key";
+  generateHmac = cfg.hmacKeyFile == null;
+
   serviceConfig = {
     systemd.services.invidious = {
       description = "Invidious (An alternative YouTube front-end)";
       wants = [ "network-online.target" ];
       after = [ "network-online.target" ];
       wantedBy = [ "multi-user.target" ];
 
+      preStart = lib.optionalString generateHmac ''
+        if [[ ! -r "${finalHmacKeyFile}" ]]; then
```

```suggestion
        if [[ ! -e "${finalHmacKeyFile}" ]]; then
```
The check for it being readable is not necessary, this file should always be readable.